src/test/java/com/google/step/youtube/DescriptionParserTest.java

```diff
@@ -33,141 +34,173 @@ public void parse_matchesFromNone() {
+        String snippetFUNGBROS10 = "Use code FUNGBROS10 ";
+        String desc = snippetFUNGBROS10 + "\nRyan: https://www.instagram.com/ryan.w.benson/";
+        OfferSnippet expected = OfferSnippet.create("FUNGBROS10", snippetFUNGBROS10);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
     }
 
     @Test
-    public void parse_codeWithQuotes() {
-        String desc = "Save 33% on your first Native Deodorant Pack - normally $36, you'll get "
-                + "it for $24! Click here: https://bit.ly/nativecoolirpa and use my code \"COOLIRPA\". ";
-        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList("COOLIRPA")));
+    public void parse_codeWithQuotes_oneLineDescription() {
+        String desc = "Click here and use my code \"COOLIRPA\". ";
+        OfferSnippet expected = OfferSnippet.create("COOLIRPA", desc);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
     }
 
     @Test
-    public void parse_toAtLink() {
-        String desc = "Go to https://NordVPN.com/pewdiepie and"
-                + "get 70% off a 3 year plan plus 1 additional month free.";
-        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList("https://NordVPN.com/pewdiepie")));
+    public void parse_toAtLink_extraLineBefore() {
+        String snippetNordVPN = "Go to https://NordVPN.com/pewdiepie and get 70% off";
+        String desc = "ty for 10 years of pewdiepie youtube uploads brvus\n" + snippetNordVPN + "\n";
+        OfferSnippet expected = OfferSnippet.create("https://NordVPN.com/pewdiepie", snippetNordVPN);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
     }
 
+    @Test
+    public void parse_repeatedPromoCodesDifferentSnippets() {
+        String firstCOOLIRPA = "Use my code \"COOLIRPA\".";
+        String secondCOOLIRPA = "Also my code \"COOLIRPA\"";
+        String desc = firstCOOLIRPA + "\n" + secondCOOLIRPA + "\n";
+
+        OfferSnippet firstExpected = OfferSnippet.create("COOLIRPA", firstCOOLIRPA);
+        OfferSnippet secondExpected = OfferSnippet.create("COOLIRPA", secondCOOLIRPA);
+        
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(firstExpected, secondExpected)));
+    }
+
+    @Test
+    public void parse_snippetTruncatedAtMaxLength() {
+        String truncatedSnippet = "App store). This episode originally recorded June 8, 2020, and "
+                + "is sponsored by Stamps.com (Go to http://stamps.com, click on the microphone "
+                + "at the top of the homepage, and type in ROOSTER to claim";
+        String desc = "Mercari (Buy or sell almost anything on Mercari on the " 
+                + truncatedSnippet + " your special offer).";
+        OfferSnippet expected = OfferSnippet.create("http://stamps.com", truncatedSnippet);
+        assertThat(DescriptionParser.parse(desc), equalTo(Arrays.asList(expected)));
```

Comment:
your assertion has logic on both sides.

go/tott/338

can you add the link to the step folder? (tott subfolder)

Reply:
Added. Does inlining the expected value remove the logic? I'm assuming the logic comes from calling the `expected` variable?